### DISCUSSION TITLE
What is meant by "multi-processing" for CumulativeMetric?

### DISCUSSION THREAD

**Discussion by ebrahimebrahim** [Project-MONAI/MONAI#4018] [Q&A] [answered]
I'm looking at the docs here: https://docs.monai.io/en/stable/metrics.html#cumulative

It says:
> In multi-processing, PyTorch programs usually distribute data to multiple nodes. Each node runs with a subset of the data, adds values to its local buffers. Calling get_buffer could gather all the results and aggregate can further handle the results to generate the final outcomes.

What's meant by "multi-processing" here? In the docs, there's an example of "maintaining two internal buffers"... in what setting is this useful?

Thanks!

**Comment by wyli** [comment, marked as answer]
it's useful when computing metrics across large datasets in a 'distributed' manner... please see this example https://github.com/Project-MONAI/tutorials/blob/master/modules/compute_metric.py

**Reply by ebrahimebrahim** [reply]
Perfect example, thank you
